### ISSUE TITLE
AutoUpgrade fail with permission issue

### ISSUE THREAD

**Issue by savitaashture** [tektoncd/operator#299]
# Expected Behavior
Operator Upgrade should work seamlessly from old version to new version
# Actual Behavior
Operator Upgrade is not working
# Steps to Reproduce the Problem

1. Apply Operator release v0.21.0-1 
2. Apply `Config` CR 
```
kubectl apply -f https://raw.githubusercontent.com/tektoncd/operator/main/config/crs/kubernetes/config/all/operator_v1alpha1_config_cr.yaml 
```
3. Apply Operator release v0.22.0-2
  
# Result
Upgradation fail with below error

```
kubectl get TektonTrigger trigger -o yaml


Install failed with message: clusterroles.rbac.authorization.k8s.io "tekton-triggers-admin" is forbidden: user "system:serviceaccount:tekton-operator:tekton-operator" (groups=["system:serviceaccounts" "system:serviceaccounts:tekton-operator" "system:authenticated"]) is attempting to grant RBAC permissions not currently held:
      {APIGroups:["serving.knative.dev"], Resources:["*"], Verbs:["get" "list" "create" "update" "delete" "deletecollection" "patch" "watch"]}
      {APIGroups:["serving.knative.dev"], Resources:["*/finalizers"], Verbs:["get" "list" "create" "update" "delete" "deletecollection" "patch" "watch"]}
      {APIGroups:["serving.knative.dev"], Resources:["*/status"], Verbs:["get" "list" "create" "update" "delete" "deletecollection" "patch" "watch"]}
```

# Additional Info

- Kubernetes version:

  **Output of `kubectl version`:**

```
Client Version: version.Info{Major:"1", Minor:"20", GitVersion:"v1.20.2", GitCommit:"faecb196815e248d3ecfb03c680a4507229c2a56", GitTreeState:"clean", BuildDate:"2021-01-13T13:28:09Z", GoVersion:"go1.15.5", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"19", GitVersion:"v1.19.2", GitCommit:"f5743093fd1c663cb0cbc89748f730662345d44d", GitTreeState:"clean", BuildDate:"2020-09-16T13:32:58Z", GoVersion:"go1.15", Compiler:"gc", Platform:"linux/amd64"}
```

**Comment by savitaashture** [comment]
Hey @vdemeester @houshengbo 

I have tried upgrading from `v0.22.0-3` to `v0.23.0-1`  and it works without any issue

I assume the rbac issue  is with older version

So do we need to fix that for that particular version :thinking:

**Comment by vdemeester** [comment]
I fixed the rbac issue on v0.22.0-3 and thus on v0.23.0-1 too yes. User should upgrade to v0.22.0-3.
/close
